packages/react/src/components/Tile/Tile.tsx

```diff
@@ -497,48 +502,41 @@ export const SelectableTile = React.forwardRef<
     className
   );
 
-  function handleClick(evt) {
+  // Single function to handle selection changes
+  const handleSelectionChange = useCallback(
+    (evt: React.SyntheticEvent, newSelected: boolean) => {
+      setIsSelected(newSelected);
+      onChange(evt as React.ChangeEvent<HTMLDivElement>, newSelected, id);
+    },
+    [onChange, id]
+  );
+
+  function handleClick(evt: React.MouseEvent<HTMLDivElement>) {
     evt.preventDefault();
     evt?.persist?.();
     if (
       normalizedDecorator &&
       decoratorRef.current &&
-      decoratorRef.current.contains(evt.target)
+      decoratorRef.current.contains(evt.target as Node)
     ) {
       return;
     }
-    setIsSelected((prevSelected) => {
-      const newSelected = !prevSelected;
-      onChange(evt, newSelected, id);
-      return newSelected;
-    });
+
+    const newSelected = !isSelected;
+    handleSelectionChange(evt, newSelected);
     clickHandler(evt);
   }
 
-  function handleKeyDown(evt) {
+  function handleKeyDown(evt: React.KeyboardEvent<HTMLDivElement>) {
```

Comment:
`KeyboardEvent` is already imported.

```suggestion
  function handleKeyDown(evt: KeyboardEvent<HTMLDivElement>) {
```